referenceframe/frame_system_test.go

```diff
@@ -237,3 +241,182 @@ func TestConvertTransformProtobufToFrameSystemPart(t *testing.T) {
+	// ------
+	name0 = "frame0"
+	pose0 = spatial.NewPoseFromPoint(r3.Vector{-4, -4, -4})
+	dims0 = r3.Vector{2, 2, 2}
+	geomCreator0, err = spatial.NewBox(pose0, dims0, "box0")
+	test.That(t, err, test.ShouldBeNil)
+	frame0, err = NewStaticFrameWithGeometry(name0, pose0, geomCreator0)
+	test.That(t, err, test.ShouldBeNil)
+	fs.AddFrame(frame0, nil)
+	// -----
+	name1 = "frame1"
+	pose1 = spatial.NewPoseFromPoint(r3.Vector{2, 2, 2})
+	dims1 = r3.Vector{2, 2, 2}
+	geomCreator1, err = spatial.NewBox(pose1, dims1, "box1")
+	test.That(t, err, test.ShouldBeNil)
+	frame1, err = NewStaticFrameWithGeometry(name1, pose1, geomCreator1)
+	test.That(t, err, test.ShouldBeNil)
+	fs.AddFrame(frame1, frame0)
+	// -----
+	inputs = StartPositions(fs)
+	_, err = FrameSystemToPCD(fs, inputs)
+	test.That(t, err, test.ShouldBeNil)
+
+	// --------------------------------------------
+	// here we are testing an arm with a block attached to the end effector
+	fs = NewEmptySimpleFrameSystem("test")
+	logger := golog.NewTestLogger(t)
+	jsonData, err := os.ReadFile(rdkutils.ResolveFile("config/data/model_frame_geoms.json"))
+	test.That(t, err, test.ShouldBeNil)
+	model, err := UnmarshalModelJSON(jsonData, "")
+	test.That(t, err, test.ShouldBeNil)
+	orientConf, err := spatial.NewOrientationConfig(spatial.NewZeroOrientation())
+	test.That(t, err, test.ShouldBeNil)
+	lc := &LinkConfig{
+		ID:          "test",
+		Parent:      "world",
+		Translation: r3.Vector{1, 2, 3},
+		Orientation: orientConf,
+	}
+	lif, err := lc.ParseConfig()
+	test.That(t, err, test.ShouldBeNil)
+	// fully specified part
+	part := &FrameSystemPart{
+		FrameConfig: lif,
+		ModelFrame:  model,
+	}
+	armFrame, _, err := CreateFramesFromPart(part, logger)
+	test.That(t, err, test.ShouldBeNil)
+	fs.AddFrame(armFrame, fs.World())
+	blockName := "block"
+	blockPose := spatial.NewPoseFromPoint(r3.Vector{2, 2, 2})
+	blockdims := r3.Vector{10, 10, 10}
+	blockGeomCreator, err := spatial.NewBox(blockPose, blockdims, "box1")
+	test.That(t, err, test.ShouldBeNil)
+	blockFrame, err := NewStaticFrameWithGeometry(blockName, pose1, blockGeomCreator)
+	test.That(t, err, test.ShouldBeNil)
+	fs.AddFrame(blockFrame, armFrame)
+	// -----
+	inputs = StartPositions(fs)
+	outMap, err = FrameSystemToPCD(fs, inputs)
+	test.That(t, err, test.ShouldBeNil)
+
+	out := outMap["test"]
+	out = append(out, outMap["block"]...)
+
+	// 1. convert 3D array to 1D
+	total := []float64{}
+	for i := range out {
+		total = append(total, out[i].X)
+		total = append(total, out[i].Y)
+		total = append(total, out[i].Z)
+	}
+	// 2. sort that
+	sort.Float64s(total)
+	// 3. Encode the value
+	var network bytes.Buffer        // Stand-in for a network connection
+	enc := gob.NewEncoder(&network) // Will write to network.
+	err = enc.Encode(total)
+	test.That(t, err, test.ShouldBeNil)
+	// 4. Hash the bytes
+	asBytes := md5.Sum(network.Bytes())
+	checkAgainst := [16]uint8{70, 188, 150, 67, 13, 145, 69, 79, 192, 180, 155, 249, 175, 57, 76, 7}
```

Comment:
I think I may have done something incorrectly with the way this check is set up. I thought that converting the 3D array to a 1D array and then sorting would ensure reproducible results, but I just got a flaky test.. LMK what you think..
<img width="1015" alt="Screen Shot 2023-01-23 at 11 11 38 AM" src="https://user-images.githubusercontent.com/46872531/214090311-62750f24-07e8-4497-b07a-72ec4ef7d8a8.png">